public/app/plugins/datasource/influxdb/components/RawInfluxQLEditor.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC } from 'react';
+import { TextArea, InlineFormLabel, Input, Select, HorizontalGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { ResultFormat, InfluxQuery } from '../types';
+import { useShadowedState } from './useShadowedState';
+import { useUniqueId } from './useUniqueId';
+
+const RESULT_FORMATS: Array<SelectableValue<ResultFormat>> = [
+  { label: 'Time series', value: 'time_series' },
+  { label: 'Table', value: 'table' },
+  { label: 'Logs', value: 'logs' },
+];
+
+const DEFAULT_RESULT_FORMAT: ResultFormat = 'time_series';
+
+type Props = {
+  query: InfluxQuery;
+  onChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+};
+
+// we handle 3 fields: "query", "alias", "resultFormat"
+// "resultFormat" changes are applied immediately
+// "query" and "alias" changes only happen on onblur
+export const RawInfluxQLEditor: FC<Props> = ({ query, onChange, onRunQuery }) => {
+  const [currentQuery, setCurrentQuery] = useShadowedState(query.query);
+  const [currentAlias, setCurrentAlias] = useShadowedState(query.alias);
+  const aliasElementId = useUniqueId();
+  const selectElementId = useUniqueId();
```

Comment:
Do we need this to be unique? I mean I know it should but feels like we could just do `influx-query-editor-select` and that would be good enough :)

Reply:
@aocenas you mean the `id`? in that case you have to be kind-of-sure that two of them cannot exist at the same time, and in the case of this widget, you can have two open at the same time in dashboard :-(